cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/SortedReplicasHelper.java

```diff
@@ -0,0 +1,151 @@
+   * @return The helper object itself.
+   */
+  public SortedReplicasHelper maybeAddPriorityFunc(Function<Replica, Integer> priorityFunc, boolean condition) {
+    if (condition && !_priorityFuncs.contains(priorityFunc)) {
+      _priorityFuncs.add(priorityFunc);
+    }
+    return this;
+  }
+
+  /**
+   * Add a priority function to the {@link SortedReplicas} to be created.
+   * Note the order adding priority function matters. Each call of this method or {@link this#maybeAddPriorityFunc(Function, boolean)}
+   * will add new priority function to the end of priority function list, which will be applied only when all the previous
+   * priority functions cannot resolve two replica's priority.
+   *
+   * @param priorityFunc The priority function to add.
+   * @return The helper object itself.
+   */
+  public SortedReplicasHelper addPriorityFunc(Function<Replica, Integer> priorityFunc) {
+    return maybeAddPriorityFunc(priorityFunc, true);
+  }
+
+  /**
+   * If condition hold, add a score function to the {@link SortedReplicas} to be created.
+   *
+   * @param scoreFunc The score function to add.
+   * @param condition Whether add the score function or not.
+   * @return The helper object itself.
+   */
+  public SortedReplicasHelper maybeAddScoreFunc(Function<Replica, Double> scoreFunc, boolean condition) {
+    if (condition) {
+      _scoreFunc = scoreFunc;
+    }
+    return this;
+  }
+
+  /**
+   * Add a score function to the {@link SortedReplicas} to be created.
+   *
+   * @param scoreFunc The score function to add.
+   * @return The helper object itself.
+   */
+  public SortedReplicasHelper addScoreFunc(Function<Replica, Double> scoreFunc) {
+    return maybeAddScoreFunc(scoreFunc, true);
+  }
+
+  /**
+   * Create {@link SortedReplicas} for given {@link Disk} with added selection/priority/score functions.
+   *
+   * @param sortName The name of {@link SortedReplicas}.
+   * @param disk The disk to create {@link SortedReplicas}.
+   */
+  public void trackSortedReplicasFor(String sortName, Disk disk) {
+    disk.trackSortedReplicas(sortName, new HashSet<>(_selectionFuncs), new ArrayList<>(_priorityFuncs), _scoreFunc);
+  }
+
+  /**
+   * Create {@link SortedReplicas} for given {@link Broker} with added selection/priority/score functions.
+   *
+   * @param sortName The name of {@link SortedReplicas}.
+   * @param broker The broker to create {@link SortedReplicas}.
+   */
+  public void trackSortedReplicasFor(String sortName, Broker broker) {
+    broker.trackSortedReplicas(sortName, new HashSet<>(_selectionFuncs), new ArrayList<>(_priorityFuncs), _scoreFunc);
+  }
+
+  /**
+   * Create {@link SortedReplicas} for each brokers in the given {@link ClusterModel} with added selection/priority/score functions.
+   *
+   * @param sortName The name of {@link SortedReplicas}.
+   * @param clusterModel The cluster model to create {@link SortedReplicas}.
+   */
+  public void trackSortedReplicasFor(String sortName, ClusterModel clusterModel) {
```

Comment:
Ideally, all `trackSortedReplicasFor` functions should be using one generic interface (e.g. with the help of generic types). It is OK to leave this to a separate patch.